tests/console/libssh.pm

```diff
@@ -0,0 +1,151 @@
+
+    #Create key pair on host and container,  import sshd host key from container, copy pub key to each other
+    my $container_ip = script_output("docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' libssh_container");
+    assert_script_run("ssh-keygen -t rsa -N '' -m pem -f /root/.ssh/id_rsa <<< y");    #Current libssh2 only support pem file format
+    assert_script_run("ssh-keyscan $container_ip >> /root/.ssh/known_hosts");
+
+    assert_script_run("docker exec libssh_container ssh-keygen -t rsa -N '' -m pem -f /root/.ssh/id_rsa <<< y");
+    assert_script_run("docker cp /root/.ssh/id_rsa.pub libssh_container:/root/.ssh/authorized_keys");
+    assert_script_run("docker cp libssh_container:/root/.ssh/id_rsa.pub /root/.ssh/authorized_keys");
+
+    #Switch into container as client
+    type_string("docker exec -it libssh_container bash\n", wait_still_screen => 3);
+    assert_script_run("test -f /.dockerenv");                                          #verify inside container
+    assert_script_run("ssh-keyscan susetest >> /root/.ssh/known_hosts");
+    validate_script_output("curl -s sftp://susetest/tmp/test/ -u root:nots3cr3t",                sub { m/libssh_testfile/ });
+    validate_script_output("curl -s sftp://susetest/tmp/test/libssh_testfile -u root:nots3cr3t", sub { m/libssh_testcase001/ });
+    assert_script_run("curl -s sftp://susetest/tmp/test/libssh_block.raw -u root:nots3cr3t -o /tmp/libssh_block.raw");
+    validate_script_output("curl -s scp://susetest/tmp/test/libssh_testfile -u root:nots3cr3t",             sub { m/libssh_testcase001/ });
+    validate_script_output('curl -s sftp://root@susetest/tmp/test/libssh_testfile --key /root/.ssh/id_rsa', sub { m/libssh_testcase001/ });
+    validate_script_output('curl -s scp://root@susetest/tmp/test/libssh_testfile --key /root/.ssh/id_rsa',  sub { m/libssh_testcase001/ });
+    validate_script_output('virsh -c "qemu+libssh://root@susetest/system?sshauth=privkey&keyfile=/root/.ssh/id_rsa&known_hosts=/root/.ssh/known_hosts" hostname', sub { m/susetest/ }) if is_sle('>=15'); #libssh is not supported by libvirt for sle12
+    validate_script_output('virsh -c "qemu+libssh2://root@susetest/system?sshauth=privkey&keyfile=/root/.ssh/id_rsa&known_hosts=/root/.ssh/known_hosts" hostname', sub { m/susetest/ });
+    #Switch back to host
+    type_string("exit\n", wait_still_screen => 3);
+    #libssh2 test with qemu-block-ssh
+    assert_script_run("eval `ssh-agent` && ssh-add /root/.ssh/id_rsa");
+    assert_script_run("qemu-system-x86_64 -daemonize -display none -drive format=raw,if=virtio,index=1,file=ssh://root\@$container_ip/tmp/libssh_block.raw -monitor unix:/tmp/socket01,server,nowait");
+    sleep 30;
+    validate_script_output('nc -U /tmp/socket01 <<EOF
+info block virtio1
+quit
+EOF
+', sub { m/libssh_block\.raw/ });
+
+    assert_script_run("docker stop libssh_container");
+}
+
+sub test_flags {
+    return {always_rollback => 1};
```

Comment:
Why is this in here? This means that this test won't work for example on bare metal or on public cloud.

Reply:
This test installed many packages i.e libvirt, qemu ... and created container. It's difficult to clean up by coding,  I used always_rollback to clean up them all so as not to affect or break the tests after it.

Reply:
I documented testcases for libssh in testopia so reviewers can better understand the test to save time. 
https://bugzilla.suse.com/tr_show_case.cgi?case_id=1768668

Reply:
OK. Then just please keep in mind that this won't work on bare metal as well as on public cloud.